src/main/java/microsoft/exchange/webservices/data/property/complex/time/TimeZoneDefinition.java

```diff
@@ -114,13 +115,18 @@ public int compare(final TimeZoneTransition x, final TimeZoneTransition y) {
     if (x == y) {
       return 0;
     } else if (x != null && y != null) {
-      final AbsoluteDateTransition firstTransition = (AbsoluteDateTransition) x;
-      final AbsoluteDateTransition secondTransition = (AbsoluteDateTransition) y;
+      if (x instanceof AbsoluteDateTransition && y instanceof AbsoluteDateTransition) {
+        final AbsoluteDateTransition firstTransition = (AbsoluteDateTransition) x;
+        final AbsoluteDateTransition secondTransition = (AbsoluteDateTransition) y;
 
-      final Date firstDateTime = firstTransition.getDateTime();
-      final Date secondDateTime = secondTransition.getDateTime();
+        final Date firstDateTime = firstTransition.getDateTime();
+        final Date secondDateTime = secondTransition.getDateTime();
 
-      return firstDateTime.compareTo(secondDateTime);
+        return firstDateTime.compareTo(secondDateTime);
+
+      } else if (y instanceof TimeZoneTransition) {
+        return 1;
```

Comment:
Yes, each Transitions in TimeZoneDefinition has one plain TimeZoneTransition as a first value. It does not have any DateTime value, so it is always "less" then any instance of AbsoluteDateTransition. In other words, it is the same case as (y == null).